hibernate-core/src/main/java/org/hibernate/boot/model/source/internal/hbm/ModelBinder.java

```diff
@@ -3494,7 +3494,20 @@ public Identifier determineImplicitName(final LocalMetadataBuildingContext conte
 
 				getCollectionBinding().setElement( elementBinding );
 
-				getCollectionBinding().setManyToManyWhere( elementSource.getWhere() );
+				final StringBuilder whereBuffer = new StringBuilder();
+				final PersistentClass referencedEntityBinding = mappingDocument.getMetadataCollector()
+						.getEntityBinding( elementSource.getReferencedEntityName() );
+				if ( StringHelper.isNotEmpty( referencedEntityBinding.getWhere() ) ) {
+					whereBuffer.append( referencedEntityBinding.getWhere() );
+				}
+				if ( StringHelper.isNotEmpty( elementSource.getWhere() ) ) {
+					if ( whereBuffer.length() > 0 ) {
+						whereBuffer.append( " and " );
+					}
+					whereBuffer.append( elementSource.getWhere() );
+				}
```

Comment:
I think you need to add extra parentheses around the conditions. Otherwise you could easily be fooled with `OR` conditions.

AFAICS, the test does not exercise the case when we have 2 conditions. It would be nice to have a test for that.

Reply:
@gsmet, I've come up with a test that reproduces incorrect results due to missing parentheses. This problem affects mapping with annotations as well as hbm.xml.

Since HHH-12875 specifically affects hbm.xml, I'll creat a new jira issue to deal with the parentheses for both annotations and hbm.xml

Reply:
I've created https://hibernate.atlassian.net/browse/HHH-12882 to add the parentheses.